devops/scripts/register_bundle_with_api.sh

```diff
@@ -159,26 +172,39 @@ else
     usage
   fi
 
-  case "${BUNDLE_TYPE}" in
+  case "${bundle_type}" in
     ("workspace") tre_get_path="api/workspace-templates" ;;
     ("workspace_service") tre_get_path="api/workspace-service-templates" ;;
     ("user_resource") tre_get_path="/api/workspace-service-templates/${workspace_service_name}/user-resource-templates";;
+    ("shared_service") tre_get_path="/api/shared-service-templates";;
   esac
 
   echo -e "Server Response:\n"
   eval "curl -X 'POST' ${tre_url}/${tre_get_path} -H 'accept: application/json' -H 'Content-Type: application/json' -H 'Authorization: Bearer ${access_token}' -d '${payload}' ${options}"
   echo -e "\n"
 
-  if ${verify}
-  then
+  if [[ "${verify}" = "true" ]]; then
     # Check that the template got registered
     template_name=$(yq eval '.name' porter.yaml)
     status_code=$(curl -X "GET" "${tre_url}/${tre_get_path}/${template_name}" -H "accept: application/json" -H "Authorization: Bearer ${access_token}" ${options} -s -w "%{http_code}" -o /dev/null)
 
-    if [[ ${status_code} != 200 ]]
-    then
+    if [[ ${status_code} != 200 ]]; then
       echo "::warning ::Template API check for ${bundle_type} ${template_name} returned http status: ${status_code}"
       exit 1
     fi
   fi
+
+  if [[ "${deploy_shared_service}" = "true" ]]; then
```

Comment:
as @tamirkamara said - this wants to have a polling loop in it to block until the shared service is deployed. it's in the e2e test code, in python - would just need to replicated here in bash

Reply:
That's a good point. I'll look into implementing that now

Reply:
This becomes too complicated for a single script. I meant to write this even before but probably forgot... IMO we should mix the register work with the deploy one and prefer to split this into another script.

Reply:
I agree with that but splitting won't be trivial (e.g. I will need to extract retrieving of the access token into a separate script, and make the new script work with any type of bundle). 
I will leave a comment here with a link to a new issue instead.